pallets/kilt-launch/src/lib.rs

```diff
@@ -0,0 +1,797 @@
+	/// One of these would be `pallet_vesting::vest()` which has to be called
+	/// actively to unlock more of the vested funds.
+	fn migrate_vesting(source: &T::AccountId, target: &T::AccountId) -> Result<Weight, DispatchError> {
+		if let Some(source_vesting) = Vesting::<T>::take(source) {
+			// Check for an already existing vesting schedule for the target account
+			// which would be the case if the claimer requests migration from multiple
+			// source accounts to the same target
+			let vesting = if let Some(VestingInfo {
+				locked,
+				per_block,
+				starting_block,
+			}) = Vesting::<T>::take(&target)
+			{
+				// Should never throw because all source accounts start vesting in genesis block
+				ensure!(
+					starting_block == source_vesting.starting_block,
+					Error::<T>::ConflictingVestingStarts
+				);
+				VestingInfo {
+					// We can simply sum `locked` and `per_block` because of the above requirement
+					locked: locked.saturating_add(source_vesting.locked),
+					per_block: per_block.saturating_add(source_vesting.per_block),
+					starting_block,
+				}
+			}
+			// If vesting hasn't been set up for target account, we can default to the one of the source
+			// account
+			else {
+				source_vesting
+			};
+			Vesting::<T>::insert(target, vesting);
+			// Only lock funds from now until vesting expires.
+			// Enables the user to have funds before actively calling `vest` if claimed
+			// after the genesis block.
+			//
+			// Logic was taken from pallet_vesting.
+
+			// Disallow transfers and reserves from vested tokens which are still locked
+			let reasons = WithdrawReasons::TRANSFER | WithdrawReasons::RESERVE;
+			let now = <frame_system::Pallet<T>>::block_number();
+			let locked_now = vesting.locked_at::<<T as pallet_vesting::Config>::BlockNumberToBalance>(now);
+			<<T as pallet_vesting::Config>::Currency as LockableCurrency<T::AccountId>>::set_lock(
+				VESTING_ID, target, locked_now, reasons,
+			);
+			Self::deposit_event(Event::AddedVesting(target.clone(), vesting.per_block, vesting.locked));
+			Ok(<T as pallet::Config>::WeightInfo::migrate_genesis_account_vesting())
+		} else {
+			Ok(T::DbWeight::get().reads(1))
+		}
+	}
+
+	/// Set the KILT balance lock for the target address which should
+	/// always be a user-owned account address.
+	///
+	/// Can be called during the migration of unowned "genesis" addresses to
+	/// user-owned account addresses in `migrate_user` as well when an account
+	/// wants to transfer their locked tokens to another account in
+	/// `locked_transfer`.
+	fn migrate_kilt_balance_lock(
+		source: &T::AccountId,
+		target: &T::AccountId,
+		// Only used for `locked_transfer`, e.g., it is `None` for migration
+		max_amount: Option<<T as pallet_balances::Config>::Balance>,
+	) -> Result<Weight, DispatchError> {
+		if let Some(LockedBalance::<T> {
+			block: unlock_block,
+			amount: source_amount,
+		}) = <BalanceLocks<T>>::get(&source)
```

Comment:
Maybe we could do something like this?

```rust
    let inner = if let Some(i) = opt {
        i
    } else {
        return Ok(T::DbWeight::get().reads(1));
    };
```

Reply:
This is the best that worked for me: d35cf33638fbdd99eca025fc03f817093b2e438e

Reply:
I actually meant something like this 🙂 : 299101b7b72f92c4182623d9615402e6ebe5ee38

Now that I touched the code, thought that we maybe want to throw an error in case the lock doesn't exists?

Reply:
If we do want to throw, we have to add an input parameter to `migrate_user`. Because when calling this function, it is not assumed that either `vesting` or `locking` exists for the source, it could just migrate the balance.

Reply:
> I actually meant something like this 🙂 : [299101b](https://github.com/KILTprotocol/mashnet-node/commit/299101b7b72f92c4182623d9615402e6ebe5ee38)

Looks good, thanks!